### PR TITLE
Net context locking missing for IPv6, CAN and packet sockets

### PR DESCRIPTION
**Pull request by jukkar**
While investigating #29444 I noticed one locking missing from net_context. Also added some way to debug locking in net_context.c and tcp2.c

### CODE REVIEW

File: subsys/net/ip/net_private.h [hunk 1]
```diff
@@ -272,3 +272,20 @@ static inline void net_pkt_print_buffer_info(struct net_pkt *pkt, const char *st
...
+#define MUTEX_LOCK(mtx) do {						\
+		NET_DBG("%d: lock %p", __LINE__, mtx);			\
+		k_mutex_lock(mtx, K_FOREVER);				\
+	} while (0)
+#define MUTEX_UNLOCK(mtx) do {						\
+		NET_DBG("%d: unlock %p", __LINE__, mtx);		\
+		k_mutex_unlock(mtx);					\
+	} while (0)
+#else
+#define MUTEX_LOCK(mtx) k_mutex_lock(mtx, K_FOREVER)
```
Comment (tbursztyka) [inline subsys/net/ip/net_private.h:289]: Since this is specific so net stack, add the prefix NET_

File: subsys/net/ip/net_private.h [hunk 2]
```diff
@@ -272,3 +272,20 @@ static inline void net_pkt_print_buffer_info(struct net_pkt *pkt, const char *st
 
 	printk("\n");
 }
+
+/* Only enable mutex lock/unlock debugging when needed as it will print
+ * lot of data which is not normally needed.
+ */
+#if 0
```
Comment (rlubos) [inline subsys/net/ip/net_private.h:279]: Is it acceptable to have it disabled/enabled like this, or shall it rather be controlled by some new Kconfig option?

**Review by rlubos — COMMENTED** [review]
LGTM, just one question

File: subsys/net/ip/net_private.h (hunk 2 above)
Comment (jukkar) [reply subsys/net/ip/net_private.h:279]: These kind of if's are quite rare, I had one in `subsys/net/lib/websocket/websocket.c`. I am not sure we want to expose this to kconfig, I would imagine that this option would be enabled in very rare cases. Enabling the debugging permanently prints too much data without much benefit.

Comment (rlubos) [reply subsys/net/ip/net_private.h:279]: Ok, understood 👍

**Review by rlubos — APPROVED** [review]

**Comment by jukkar** [comment]
@pfalcon can you help reviewing this PR, Tomasz is on vacation this week and I would like to get this merged sooner than later.

**Review by pfalcon — APPROVED** [review]

**Pull request merged**